### ISSUE TITLE
Whoops! An error occurred

### ISSUE THREAD

**Issue by cthmpsn** [firefly-iii/firefly-iii#3110]
I am running Firefly III version 4.8.2

**Description**
I had installed Firefly III on a VM Centos 8 server using Apache and PHP 7.4. I was able to create and configure a virtual host to listen on 443. This is self hosted, and I am only able to contact the VM from my laptop using VirtualBox host only adapter.

Before I did updates to CentOS on 1-25-2020, the site worked and was able to connect. After the installing the updates, the site was down. However, I can connect to the Apache server via port 80.
I troubleshot the problem and couldn't come up with a solution. So I left it alone for a couple of weeks.

Today, I go into the VM to work on the issue, and still can't connect via port 443. So I change the .conf file to have the virtual host run on port 80. When I try to access the firefly web page, I get the login page. Then after logging in, I receive the "Whoops! An error occured.." screen with this as the error:

> Access level to PragmaRX\Google2FALaravel\Support\Config::config() must be public (as in class PragmaRX\Google2FALaravel\Support\Auth)

For some reason I couldn't add a screenshot. 

My main concern is that I would like to get a copy or backup the data. I don't have any problems reinstalling firefly if I have to (especially that I see that version 5.0 is out)

**Extra info**

Debug information generated at 2020-02-10 13:20:25 America/New_York for Firefly III version **4.8.2**.

| Variable | Content |
| --- | --- |
| FF version | 4.8.2 |
| FF API version | 0.10.5 |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | , stdout |
| PHP version | 7.4.2 |
| Display errors | Off |
| Session start |  |
| Session end |  |
| Session first |  |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | fpm-fcgi |
| UserID | 1 |
| Attempt at "en" | false |
| Attempt at "English" | false |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, sqlite |
| Current driver | mysql |
| Login provider |   |
| Storage disks | local-upload  |
| Using Sandstorm? | no |
| Is Sandstorm (.env) | false |
| bunq uses sandbox | false |
| Trusted proxies (.env) |  |
| User agent | Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0 |
| Loaded extensions | Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, cgi-fcgi, bcmath, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, intl, json, ldap, exif, mysqlnd, PDO, Phar, SimpleXML, sockets, sodium, sqlite3, tokenizer, xml, xmlwriter, xsl, mysqli, pdo_mysql, pdo_sqlite, xmlreader, zip, Zend OPcache |

**Bonus points**

- [X] I have read the FAQ at https://bit.ly/FF3-FAQ
- [ ] I added a screenshot
- [X] I added log files (see https://bit.ly/FF3-get-logs)
- [ ] I was able to replicate the issue on the demo site.

**Comment by JC5** [comment]
Try running “composer install” in your Firefly III directory to make sure all your libraries are updated as well.

**Comment by irateadmin** [comment]
I have this same error but I am using CentOS 7 and nginx. A composer install did not fix the issue. I intend to see if an upgrade to v5.0.4 fixes this issue or not.

**Comment by irateadmin** [comment]
Wanted to followup and say that I don't have this issue after upgrading to 5.0.4.

**Comment by cktsc** [comment]
@irateadmin that's what I was looking at doing anyhow. Did upgrading preserve your data that was already entered into the server?

**Comment by irateadmin** [comment]
@cktsc - The upgrade process preserves your data. I followed the instructions on this page: https://docs.firefly-iii.org/advanced-installation/upgrade

If you have SELinux enabled or data in different locations, you will have to adjust accordingly.

Reach out if you need any help. I recommend the Veeam free Linux agent for backups if you want to make sure you have a complete rollback plan.

**Comment by cktsc** [comment]
@irateadmin Thanks. I will work on the upgrade this week. I will report back with my results.

**Comment by ChrisCarini** [comment]
I'm also experiencing this. Any guidance would be appreciated!

**Comment by irateadmin** [comment]
> I'm also experiencing this. Any guidance would be appreciated!

I fixed this issue by following this guide to upgrade to the latest version.
https://docs.firefly-iii.org/advanced-installation/upgrade

**Comment by ChrisCarini** [comment]
Upgrading is not always a trivial task for me, so I like exploring what other options exist (if any) first.

**Comment by Lawri-van-Buel** [comment]
@ChrisCarini what do you mean with non trivial task?

Firefly is quite easy to update in its current form and @JC5 does a good job in my books in keeping it simple and easy. 

If you need specific knowledge or understanding we need to know where you are either lacking knowledge or where the instructies or not clear enough. Either will be a point we can assist or improve. (We all had to learn it somewhere so not knowing is not a shame). I hope this helps in getting better wensers.

**Comment by ChrisCarini** [comment]
Naa, the firefly-iii docs are totally fine. It's 100% a "me" problem with my setup.

Tl;dr, whenever I upgrade, I have to compile a custom build of PHP and make tweaks here and there in firefly code due to some awesome memory constraints. Normally with your docs it'd probably take my a few minutes, but I typically end up spending a lot longer. :(

I upgraded last night to `5.0.5` and I don't have the issue any more. :) Happy days!

**Comment by Lawri-van-Buel** [comment]
Might I put a few suggestions your way to look at to make it easier for you. 
Have you considered :
- using docker
- using lxc / lxd containers (quite east to use and setup on a single system and would allow a dedicated separated setup of php directly from a repository)
- using ansible to automate your process in order to automate your workflow. 

Good luck with it.

**Comment by JC5** [comment]
And send me those code changes, you’ve made me curious 😁

**Comment by ChrisCarini** [comment]
@Lawri-van-Buel Yeaaahh... Considered all of those; sadly this is in shared hosting so they aren't options :( :(

I've considered moving this into my home network, but I have other users of firefly that aren't technical enough to have VPN setup for access (I don't expose anything directly on my home network, self policy)

@JC5 yeah I can totally grab the exact file(s) and line numbers next time! Fwiw, it's inconsistent between upgrades (even same version) because sometimes there is enough memory available to my account on the shared hosting, sometimes not. I do recall one/some of them are due to memory allocation for regex - the 'fix' there is luckily a one-liner addition.
